src/Servers/Kestrel/shared/KnownHeaders.cs

```diff
@@ -1328,6 +1328,15 @@ public unsafe bool TryHPackAppend(int index, ReadOnlySpan<byte> value, bool chec
             }}
         }}
 
+        [MethodImpl(MethodImplOptions.AggressiveOptimization | MethodImplOptions.AggressiveInlining)]
+        public void MergeCookies()
+        {{
+            if ((_bits & 0x20000L) != 0 && _headers._Cookie.Count > 1)
```

Comment:
The bits value for cookie could change and this if will break.

There's an example of capturing bit values here: https://github.com/dotnet/aspnetcore/blob/dc98f4b9580fde664c6ae88f9712467ff8e96bc4/src/Servers/Kestrel/shared/KnownHeaders.cs#L263-L266

You don't need to aggregate a bunch of bit values together. You just need to get the cookies one and use it here.

Reply:
Even better, there is already a pattern for this: https://github.com/dotnet/aspnetcore/blob/dc98f4b9580fde664c6ae88f9712467ff8e96bc4/src/Servers/Kestrel/Core/src/Internal/Http/HttpHeaders.Generated.cs#L385-L386

Add cookies to this collection and then use generated HasCookies property: https://github.com/dotnet/aspnetcore/blob/c1febaecce1a5844d2d98ec87e97279d35ff7ac4/src/Servers/Kestrel/shared/KnownHeaders.cs#L102-L106

Reply:
> You don't need to aggregate a bunch of bit values together. You just need to get the cookies one and use it here.
Sorry, can you please explain what you mean by this? I'm not quite sure what you mean by just getting the cookie one? This is the pattern that @halter73 suggested. Also, in your second comment, do you mean just add a new `public bool` for this pattern?

Reply:
> Sorry, can you please explain what you mean by this? 

Ignore first comment. The second comment explains how it should be done.

0x20000L is generated when code generation runs. When the order of properties changes then the bit value for cookie will change. You've hardcoded it in this method so the method will stop working correctly.

You need to use a bit value that is known for cookie. A generated public property will react to bit value changing.